shared/js/moz_intl.js

```diff
@@ -187,36 +199,257 @@ global.mozIntl = {
     return navigator.mozL10n.formatValue('timePattern_' + type).then(fmt => {
       return {
         format: function(input) {
-          var hour, min, sec, ms;
-          var totalTime = input;
-
-          hour = Math.floor(totalTime / (1000 * 60 * 60));
-          totalTime -= hour * (1000 * 60 * 60);
-          min = Math.floor(totalTime / (1000 * 60));
-          totalTime -= min * (1000 * 60);
-          if (type === 'hms') {
-            sec = Math.round(totalTime / 1000);
-          } else {
-            sec = Math.floor(totalTime / 1000);
-            totalTime -= sec * 1000;
-            // here we take tens of a msec only
-            ms = Math.round(totalTime / 10);
+          const format = durationFormats[type];
+          const duration = splitIntoTimeUnits(input, format.max, format.min);
+
+          if (duration.hasOwnProperty('millisecond')) {
+            // We round milliseconds to 2-digit
+            duration.millisecond =
+              Math.round(duration.millisecond / 10);
           }
 
           var string = fmt;
-          if (type === 'hms') {
-            string = string.replace('hh', numFormatter.format(hour));
-          }
-          string = string.replace('mm', numFormatter.format(min));
-          string = string.replace('ss', numFormatter.format(sec));
-          if (type === 'msS') {
-            string = string.replace('SS', numFormatter.format(ms));
-          }
+
+          format.tokens.forEach(([token, unit]) => {
+            string = string.replace(token,
+              numFormatter.format(duration[unit]));
+          });
           return string;
         }
       };
     });
+  },
+
+  /**
+   * RelativeTime formatter.
+   * Formats an integer with milliseconds into locale specific relative time 
+   * string.
+   *
+   * Currently accepted options:
+   *
+   * * style - long | short
+   *
+   * Defines whether the string will be long "1 minute ago" or short "1 min. 
+   * ago"
+   *
+   * * minUnit - (default) millisecond
+   * * maxUnit - (default) year
+   * * unit - second | minute | hour | day | week | month | year
+   *          (default) bestFit
+   *
+   * Example:
+   *
+   * const formatter = new Intl.RelativeTimeFormat(navigator.languages, {
+   *   unit: 'bestFit',
+   *   style: long'
+   * });
+   *
+   * var ms = Date.now() - 2 * 1000;
+   *
+   * formatter.format(ms); // "2 seconds ago"
+   *
+   * @param {Array} An array of languages
+   * @param {Array} Options object
+   * @returns An object with a formatter that returns Promises of strings
+   */
+  RelativeTimeFormat: function(locales, options) {
+    return {
+      /*
+       * ECMA 402 rev 3., 1.3.4, FormatRelativeTime
+       *
+       * Notes: This is a modified version of the function to use mozL10n
+       * and simplified to match current data set in data.properties
+       */
+      format: function(x) {
```

Comment:
Yeah, I'm following the spec now - https://rawgit.com/caridy/intl-relative-time-spec/master/index.html 1.3.4
`4. Let x be ToNumber(value).`

I'll suggest a better name in a spec and will keep our code in line with the spec.